Bug fix: library path and missing anvil package

nturaga opened the pull request:
This PR fixes two bugs,

1. Missing library 'AnVIL'

2. The wrong library path that is used for the persistent library as shown in the screen shot.

<img width="1280" alt="Screen Shot 2022-05-03 at 4 06 13 PM" src="https://user-images.githubusercontent.com/2746443/166732995-89778864-748b-4b92-a301-74ae10b7fbc1.png">

nturaga committed:
Bug fix: Install AnVIL package in the RStudio image

- AnVIL package needs to be installed in the RStudio image
- Patch version update 3.15.1

nturaga committed:
Bug fix: R library path name issue caused by R-4.2.0

$ R CMD printenv R_LIBS_USER
`~/R/x86_64-pc-linux-gnu-library/4.1`

$ R CMD printenv R_LIBS_USER
%U

This is documented in the R documentation:

In the NEWS file there is
    • Environment variables R_LIBS_USER and R_LIBS_SITE are both now
      set to the R system default if unset or empty, and can be set to
      NULL to indicate an empty list of user or site library
      directories.

nturaga committed:
remove the "/" to fix bug

gpcarr commented on anvil-rstudio-bioconductor/scripts/set_up_package_dir.sh:
```diff
@@ -6,9 +6,10 @@ set -e
+# R_PATH=`R CMD printenv R_LIBS_USER | sed 's/~\///g'`
 BIOCONDUCTOR_VERSION=`printenv BIOCONDUCTOR_DOCKER_VERSION | sed 's/\(^[0-9].[0-9][0-9]\).*/\1/g'`
-R_PACKAGE_DIR=${RSTUDIO_USER_HOME}/${R_PATH}-${BIOCONDUCTOR_VERSION}
+R_PACKAGE_DIR=${RSTUDIO_USER_HOME}${R_PATH}-${BIOCONDUCTOR_VERSION}
```
talked about this via slack but posting here for visibility since i'm out the rest of this week

the issue is that `R_PATH` is `/home/rstudio/R/x86_64-pc-linux-gnu-library/4.2` and `RSTUDIO_USER_HOME` is `/home/rstudio` so that would mean that `R_PACKAGE_DIR=${RSTUDIO_USER_HOME}/${R_PATH}-` is `/home/rstudio//home/rstudio/R/x86_64-pc-linux-gnu-library/4.2-` when it should just be `home/rstudio/R/x86_64-pc-linux-gnu-library/4.2-`

`R_PATH` used to be `R/x86_64-pc-linux-gnu-library/4.2` so we should make sure it's still getting that value

nturaga replied:
R_PATH is actually `/root/R/x86_64-pc-linux-gnu-library/4.2`. 

The fix I issued now makes R_PATH `R/x86_64-pc-linux-gnu-library/4.2`.

gpcarr reviewed (changes requested):
see comment in `set_up_package_dir`

nturaga committed:
Bug fix:

This fixes the issue where R_PATH adds '/root' as the directory.

These lines are being run as the root user, and this will fix the
issue and make sure that if the /root is present it will be replaced
with a blank space.

nturaga commented:
There is a custom image [us.gcr.io/anvil-gcr-public/anvil-rstudio-bioconductor:nitesh-test-2](http://us.gcr.io/anvil-gcr-public/anvil-rstudio-bioconductor:nitesh-test-2), that I tested. and it works as expected.

results: 
<img width="1784" alt="Screen Shot 2022-05-06 at 11 21 14 AM" src="https://user-images.githubusercontent.com/2746443/167163079-73873e3b-941a-4807-96e2-7cf978fc00cf.png">

nturaga commented on anvil-rstudio-bioconductor/scripts/set_up_package_dir.sh:
```diff
@@ -6,11 +6,20 @@ set -e
+# with a blank space.
 R_PATH=`Rscript -e "cat(Sys.getenv('R_LIBS_USER'))"`
-# R_PATH=`R CMD printenv R_LIBS_USER | sed 's/~\///g'`
+CHECK_ROOT='/root'
```
This will fix the issue.

nturaga commented:
I don't know if @Qi77Qi is available to take a look a this? I believe the issue was misunderstood by @gpcarr .

nturaga commented:
@gpcarr Merging this because I've not heard back. Let me know if any other changes are needed.
